### PR TITLE
Created server/client side rendered component

### PR DESCRIPTION
**Pull request by fniewijk**
Created Search Products component

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,23 +8,27 @@
...
     "ignore-styles": "^5.0.1",
     "istanbul": "^0.4.5",
+    "jsdom": "^9.5.0",
     "mocha": "^3.0.2",
     "require-all": "^2.0.0"
   },
   "scripts": {
     "start": "npm-run-all bundle open:src",
-    "dev": "npm-run-all bundle dev-parallel open:src",
-    "dev-parallel": "npm-run-all --parallel lint:watch test-unit:watch sass-lint",
```
Comment (taraojo) [inline package.json]: When will the unit tests be run?

File: src/server/routes/api/handlers.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+const http = require('http');
+
+const searchProductsPromise = function(query) {
+    return new Promise((resolve, reject) => {
+
+        const options = {
+            host: 'marksandspencer-stage.apigee.net',
+            path: '/v1/search/products?searchTerm=' + query + '&apikey=GPHqj2nEVt3jeS1gyrqiLWf56hxHdNzt',
```
Comment (taraojo) [inline src/server/routes/api/handlers.js]: Are we ok to keep this key in git?

Comment (fniewijk) [inline src/server/routes/api/handlers.js]: These API keys are ok to share with the public, the API team said. We will have to get our own one though while we are in production, so they can see easily where the load comes from. You are right though that the API keys should be injected into the project via enviromental variables

File: package.json (hunk 1 above)
Comment (fniewijk) [inline package.json]: The unit tests are broken at the moment, we want to fix them in master.

File: src/client/plp.js [hunk 3]
```diff
@@ -1,5 +1,82 @@
...
+    const doApiSearch = function(searchText) {
+        const url = 'api/searchProducts?' + searchText;
+        return getJson(url);
+    };
+
+    const getSearchProductsTemplate = function(){
+        return getTemplate('assets/templates/searchProducts.html');
+    };
+
+    const getTemplate = function (url) {
```
Comment (SKSyal) [inline src/client/plp.js]: Similar function - may be we can keep only one with switch case statement.

**Pull request merged**